### ISSUE TITLE
compile rnw via knitr

### ISSUE THREAD

**Issue by maxheld83** [thomasjo/atom-latex#101]
... that'd be nice :+1:

**Comment by thomasjo** [comment]
Not having used either Rnw (Sweave) or knitr, do you have any examples for reference? Seems like a great thing to support, but I don't really have a clue where to start since I know virtually nothing about either pieces of software.

**Comment by m0nhawk** [comment]
Useful link regarding compiling [`knitr`](http://tex.stackexchange.com/questions/132704/how-to-build-knitr-document-from-the-command-line) documents. It have pretty simple and few command to compile the document.

**Comment by PMassicotte** [comment]
This would be very nice indeed :)

**Comment by thomasjo** [comment]
Would those interested in this feature mind testing out #188? The easiest way is likely to do `apm develop latex` and then running Atom in dev mode via `atom --dev`. I'm reasonably confident everything works as expected, but I might have missed something obvious 🙈

**Comment by PMassicotte** [comment]
Don't know if I missed something, but after installing the dev version and running `atom --dev` I can't compile the .Rnw file. The build is not even lunching.

**Comment by thomasjo** [comment]
@PMassicotte Do you see any errors in Atom's developer console? Also what TeX distribution and OS are you running?

Please note that as it stands, this feature requires that you have both a semi-recent installation of R, it requires that `Rscript` is discoverable via your `PATH` environment variable, and it also requires that the Knitr library is installed.

What is the output of the following commands?
``` bash
which Rscript
Rscript -e "packageVersion('knitr')"
```

Obviously you only run the second command if the first one doesn't return an error.

**Comment by PMassicotte** [comment]
Nothing happen when I try to build the `Rnw` document. 

I think I have the right installation.

```bash
which Rscript
/usr/bin/Rscript

Rscript -e "packageVersion('knitr')"
[1] ‘1.13’
```

As for my tex version:

```bash
tex --version

TeX 3.14159265 (TeX Live 2015/Debian)
kpathsea version 6.2.1
Copyright 2015 D.E. Knuth.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the TeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the TeX source.
Primary author of TeX: D.E. Knuth.
```

My OS:

```bash
lsb_release -a

No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Comment by thomasjo** [comment]
I just realized I left out a vital step; since I haven't merged this PR yet, you have to checkout the `knitr-support` branch to test this PR;

``` bash
$ cd ~/.atom/dev/packages/latex
$ git checkout knitr-support
```

Then try starting Atom in dev mode again.

**Comment by PMassicotte** [comment]
Now it is compiling a PDF but I get the following error.

```r
plot(1)
## Error in eval(expr, envir, enclos):
could not find function "plot"
```

Here is my file:

```latex
\documentclass{article}

\begin{document}


<<>>=
plot(1)
@

\end{document}
```

**Comment by thomasjo** [comment]
@PMassicotte I just pushed a commit that should resolve that problem! Turns that little refactoring I did recently caused a regression that prevented **all** of the default R packages from being loaded. 

To test the latest changes, just navigate to `~/.atom/dev/packages/latex` and run `git pull`. Hopefully this resolves the problem on your end as well.

I suppose I should add a new spec that actually verifies that the output of passing a `.Rnw` file through the Knitr produces the expected `.tex` file.
